BoostedAnalyzer/src/MCMatchVarProcessor.cpp

```diff
@@ -468,4 +598,107 @@ bool dfirst=true;
     vars.FillVar( "GenForwardQuark_E",forward_quark.energy());
     vars.FillVar( "GenForwardQuark_PDGID",forward_quark.pdgId());
   }
+
+
+    //if(input.genTopEvt.IsFilled()&&input.genTopEvt.TTxIsFilled()&&input.genTopEvt.IsSemiLepton()) {
```

Comment:
Whats happening here?

Reply:
Hello sebwieland,
thank you for your comment.

This part of the module tries to find matching reco-level jets for all objects in a ttH/ttbar sample, i.e. tries to find the b-Quarks from top decays, the light flavour quarks from the hadronic W decay and for ttH additionally both b-Quarks from the Higgs decay.
From this information the flags `Gen_ttbar/ttH_matched` are generated which can be used to generate a subsample of events where a reconstruction of ttH/ttbar is possible in general.